### ISSUE TITLE
feat: Binary cross-entropy loss function for classification tasks

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#95]
## Summary

Zerfoo's training infrastructure currently provides MSE loss for regression. For binary classification tasks (e.g., predicting directional movement: up=1, down=0), **binary cross-entropy (BCE) loss** would produce better-calibrated probability outputs.

## Current Behavior

Wolf's `ZerfooTabularBackend` uses manual MSE loss computation:

```go
lossVal, lossGrad, err := computeMSELoss(engine, ctx, output, targetTensor)
```

For binary classification with {0, 1} labels, MSE works but:
- Gradients are not optimal for classification (treats 0.4→0.5 the same as 0.9→1.0)
- Output is unbounded (can be <0 or >1), requiring post-hoc clamping
- No sigmoid activation on output layer

## Proposed API

```go
// In package training/loss or layers/loss

// BinaryCrossEntropy computes BCE loss and gradient.
// Applies sigmoid internally: loss = -[y*log(σ(x)) + (1-y)*log(1-σ(x))]
func BinaryCrossEntropy[T numeric.Float](
    engine compute.Engine[T],
    ctx context.Context,
    logits, targets *tensor.TensorNumeric[T],
) (T, *tensor.TensorNumeric[T], error)
```

Or as a `Loss` interface that pairs with `Trainer[T]`:

```go
type Loss[T numeric.Float] interface {
    Compute(ctx context.Context, predictions, targets *tensor.TensorNumeric[T]) (T, *tensor.TensorNumeric[T], error)
}

type BCEWithLogitsLoss[T numeric.Float] struct { ... }
type MSELoss[T numeric.Float] struct { ... }
```

## Use Case

Wolf trains models to predict stock price direction (binary: up=1, down=0). LightGBM (metee) uses `binary` objective internally and outputs calibrated probabilities. For a fair ensemble comparison, the neural net backend should also be trained with BCE loss and output probabilities via sigmoid.

## Additional Context

Related to the ensemble architecture where LightGBM outputs probabilities [0,1] and the MLP should output compatible probability-like scores for meaningful weighted combination.
